### PR TITLE
Use lockRole for LOCK admin

### PR DESCRIPTION
**Pull request by uivlis**
Fix for [this](https://github.com/code-423n4/2021-05-yield-findings/issues/50).

### CODE REVIEW

File: contracts/access/AccessControl.sol [hunk 1]
```diff
@@ -127,6 +127,7 @@ contract AccessControl {
      * - the caller must have ``role``'s admin role.
      */
     function setRoleAdmin(bytes4 role, bytes4 adminRole) external virtual admin(role) {
+        require(role != LOCK, "Use `lockRole`");
```
Comment (albertocuestacanada) [inline contracts/access/AccessControl.sol:130]: Test, please. We always include a test for each `require` (or we should, I know I've cut corners before).

**Review by albertocuestacanada — CHANGES_REQUESTED** [review]

File: contracts/access/AccessControl.sol (hunk 1 above)
Comment (uivlis) [reply contracts/access/AccessControl.sol:130]: Woah... After testing, I have strong reasons to believe this is not actually an issue, I will comment below.

**Comment by uivlis** [comment]
I can best explain my stance that this is not an issue by pointing to the test I just pushed, please have a look at it and let me know if it is, in fact, as self-evident as I think.

**Comment by albertocuestacanada** [comment]
The issue is that roles could be accidentally locked using `contract.setRoleAdmin(role, LOCK)`, not that someone could gain control of the LOCK role.

Note that any roles, including not-yet existing ones, can be controlled by ROOT. Only LOCK is unavailable to ROOT.

The test that now fails is the one we need to fix the code for.

**Review by albertocuestacanada — CHANGES_REQUESTED** [review]
See comment above :point_up:

**Comment by uivlis** [comment]
Ah... I see, I must have misunderstood the issue from the beginning. Fixing.

**Comment by uivlis** [comment]
Alright, I think I got it.

**Review by albertocuestacanada — APPROVED** [review]
Good to go!

**Pull request merged**